tests/integration.test.js

```diff
@@ -3,7 +3,7 @@
 const assert = require('assert');
 const fs = require('fs');
 const iiif = require('../index');
-const probe = require('probe-image-size');
```

Comment:
probe-image-size has an advantage of not needing to load the image its substantially faster for size info. however, it looks like maybe we have to load it to do processing anyway and this code may only load it one time. Am I reading this right that we are replacing a fast thing followed by a slow thing with simply doing just the slow thing?

Reply:
I did some benchmarking on sharp vs. probe for getting image sizes, and sharp turned out to be much, much faster. (I was actually kind of surprised.) Sharp doesn't have to load the entire image to get size info; it can seek to the header(s) for metadata the same way probe does. The secret sauce turned out to be passing an explicit page number to the pipeline constructor:

```javascript
const imageStream = createReadableStreamFromImageSource();
const metadataReader = sharp({ page: 0 });
imageStream.pipe(metadataReader);
const { width, height, pages } = await imageStream.metadata();
```

It's possible to read the sizes of all the pages explicitly by repeating the above process for `page` values of `1` to `pages - 1`, but it requires re-creating the stream each time so it's far more efficient to assume that each page is `Math.floor([half the size of the previous one])`, as it is in any sane pyramidal format. Implementers are always welcome to override the dimension function to use probe or look it up in a database or S3 metadata/tags or whatever.

Reply:
I'll post the actual benchmarks tomorrow morning just to have a record.

Reply:
I'm certainly content with that =-)